shell/platform/android/io/flutter/embedding/android/FlutterActivity.java

```diff
@@ -56,6 +56,14 @@
  *   <li>{@link #getDartEntrypoint()}</li>
  *   <li>{@link #getInitialRoute()}</li>
  * </ul>
+ * <p>
+ * A custom Dart entrypoint or Flutter route can also be specified by using an existing {@link
+ * FlutterEngine}. A {@link FlutterEngine} can be provided by implementing the
+ * {@link #getFlutterEngine} method. Creating a {@link FlutterEngine} before showing a {@code
```

Comment:
You should probably hold off with this explanation until my current PR is merged. I've renamed this method to `provideFlutterEngine` because it turns out there were 2 methods called `getFlutterEngine` for 2 different purposes. I've also made `FlutterActivity` implement `FlutterEngineProvider` so that devs no longer have to worry about adding that interface themselves - instead they can just override `provideFlutterEngine` in their subclass.

Reply:
Ya sure, let's get your PR through first.

Reply:
@matthew-carroll @xster is the PR in question now merged?

Reply:
Yes, the PR was merged. There have also been other non-trivial changes to `FlutterActivity` so its probably worth re-checking all language in this PR against the latest.

Reply:
Cool, I can revise this one at some point this week.